node/src/components/linear_chain_sync/operations.rs

```diff
@@ -128,12 +195,42 @@ where
         {
             Ok(FetchedData::FromStorage { item }) => return Ok(Some(item)),
             Ok(FetchedData::FromPeer { item, .. }) => {
-                // TODO: Validate item, ban peer if necessary
                 // Compute the total weight of the validators
-                let _total_weight: U512 = trusted_validator_weights
-                    .iter()
-                    .map(|(_, weight)| *weight)
-                    .sum();
+                let BlockHeaderWithMetadata {
+                    block_header,
+                    block_signatures,
+                } = *item.clone();
+
+                match validate_finality_signatures(
+                    &block_header,
+                    &trusted_validator_weights,
+                    finality_threshold_fraction,
+                    &block_signatures,
+                ) {
+                    Ok(()) => {}
+                    Err(error) => {
```

Comment:
Maybe `if let Err(error) = validate_…` would be more concise than `match`, since we don't do anything in the `Ok(())` arm.

Reply:
Fixed in 24650baf9ab7725c17151ea550dc074206c98a18